src/appindexing.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+goog.provide('appindexing');
+
+goog.require('goog.json');// jshint unused:false
+goog.require('utils');
+
+
+appindexing.state = {};
+appindexing.state['androidAppIndexingTagsPresent'] = false;
+appindexing.state['iOSAppIndexingTagsPresent'] = false;
+appindexing.state['androidDetailsComplete'] = false;
+appindexing.state['iOSDetailsComplete'] = false;
+appindexing.state['addediOSTagDom'] = false;
+appindexing.state['addedAndroidTagDom'] = false;
+
+appindexing.checkForAppIndexingTags = function() {
+	var linkTags = document.getElementsByTagName('link');
+	var numLinkTags = linkTags.length;
+
+	for (var counter = 0; counter < numLinkTags; counter++) {
+		var currLinkTag = linkTags[counter];
+		var currLinkTagHref = currLinkTag.href;
+		if (currLinkTagHref.includes('ios-app')) {
+			appindexing.state['iOSAppIndexingTagsPresent'] = true;
+			currLinkTag.setAttribute('href',  appindexing.addBranchTrackingParams(currLinkTagHref));
+		}
+		if (currLinkTagHref.includes('android-app')) {
+			appindexing.state['androidAppIndexingTagsPresent'] = true;
+			currLinkTag.setAttribute('href',  appindexing.addBranchTrackingParams(currLinkTagHref));
+		}
+	}
+};
+
+appindexing.checkOptions = function(options) {
+	if (options.hasOwnProperty('androidPackageName') && options.hasOwnProperty('androidURL') && typeof options['androidPackageName']!==undefined && typeof options['androidURL']!==undefined){
```

Comment:
can you put these different conditionals on new lines for readability and be sure to add spaces before & after `!==`